apps/web/src/pages/Users/UpdateUserPage/components/IndividualRoleTable.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useMemo } from "react";
+import { useIntl } from "react-intl";
+
+import { notEmpty } from "@gc-digital-talent/helpers";
+import { Pill } from "@gc-digital-talent/ui";
+import { getLocalizedName } from "@gc-digital-talent/i18n";
+
+import Table, { ColumnsOf, Cell } from "~/components/Table/ClientManagedTable";
+import { getFullNameHtml } from "~/utils/nameUtils";
+import { Role, UpdateUserAsAdminInput, User } from "~/api/generated";
+
+import AddIndividualRoleDialog from "./AddIndividualRoleDialog";
+import RemoveIndividualRoleDialog from "./RemoveIndividualRoleDialog";
+import { UpdateUserFunc } from "../types";
+
+const roleCell = (displayName: string) => {
+  return (
+    <Pill color="neutral" mode="solid">
+      {displayName}
+    </Pill>
+  );
+};
+
+const actionCell = (role: Role, user: User, onUpdateUser: UpdateUserFunc) => (
+  <RemoveIndividualRoleDialog
+    role={role}
+    user={user}
+    onUpdateUser={onUpdateUser}
+  />
+);
+
+type RoleCell = Cell<Role>;
+
+interface IndividualRoleTableProps {
+  user: User;
+  availableRoles: Array<Role>;
+  onUpdateUser: UpdateUserFunc;
+}
+
+export const IndividualRoleTable = ({
+  user,
+  availableRoles,
+  onUpdateUser,
+}: IndividualRoleTableProps) => {
+  const intl = useIntl();
+  const userName = getFullNameHtml(user.firstName, user.lastName, intl);
+
+  const columns = useMemo<ColumnsOf<Role>>(
+    () => [
+      {
+        Header: intl.formatMessage({
+          defaultMessage: "Actions",
+          id: "S8ra2P",
+          description: "Title displayed for the role table actions column",
+        }),
+        accessor: (d) => `Actions ${d.id}`,
+        disableGlobalFilter: true,
+        Cell: ({ row: { original: role } }: RoleCell) =>
+          actionCell(role, user, onUpdateUser),
+      },
+      {
+        Header: intl.formatMessage({
+          defaultMessage: "Role",
+          id: "uBmoxQ",
+          description: "Title displayed for the role table display name column",
+        }),
+        accessor: (role) => getLocalizedName(role.displayName, intl),
+        Cell: ({ row: { original: role } }: RoleCell) =>
+          roleCell(getLocalizedName(role.displayName, intl)),
+      },
+    ],
+    [intl, onUpdateUser, user],
+  );
+
+  // TO DO: Update to user roles
+  const data = useMemo(() => {
+    const roles = user.roleAssignments
+      ?.filter(notEmpty)
+      .map((assignment) => assignment.role)
+      .filter(notEmpty);
+    return roles || [];
+  }, [user.roleAssignments]);
```

Comment:
need to filter out team based roles so pool operator does not appear

Reply:
I see your comment

> Yup, but this is totally valid. However, this is for individual roles so this is a bug I need to resolve since no team roles should appear here.

Reply:
oh, and maybe add the headers for the individual role table section?

![image](https://user-images.githubusercontent.com/40485260/222854805-3191f553-c72f-4edf-9072-85975ffd47d8.png)

per

![image](https://user-images.githubusercontent.com/40485260/222854904-0614964c-735e-407c-8d71-7dee037d23a0.png)

Reply:
> Should individual roles disappear from the list when they're already added? If not now, should we spin that off into a separate issue? (same for removal of roles)

Removed current user roles in [filter out team based roles and current user roles](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5799/commits/0f56ac40106f72881e701ae9633a5a64a41ccb4e)

Reply:
Added headings in [add headings to roles section](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5799/commits/14e93aadab56ba2a56136a4d99460ca27288c18b)